client/apps/edit/components/section_image_collection/index.coffee

```diff
@@ -3,151 +3,55 @@
 
   componentDidMount: ->
     @$list = $(@refs.images)
-    @setupAutocomplete()
     imagesLoaded @$list, =>
-      if @state.images.length > 1 and @state.layout is 'overflow_fillwidth'
-        @toggleFillwidth()
-      else
-        @$list.animate({opacity: 1}, 500)
-
-  componentWillUnmount: ->
-    @autocomplete.remove()
-
-  componentWillUpdate: ->
-    @props.section.set images: @state.images if @state.images.length > 0
-
-  setupAutocomplete: ->
-    $el = $(@refs.autocomplete)
-    @autocomplete = new Autocomplete
-      url: "#{sd.ARTSY_URL}/api/search?q=%QUERY"
-      el: $el
-      filter: (res) ->
-        vals = []
-        for r in res._embedded.results
-          if r.type?.toLowerCase() == 'artwork'
-            id = r._links.self.href.substr(r._links.self.href.lastIndexOf('/') + 1)
-            vals.push
-              id: id
-              value: r.title
-              thumbnail: r._links.thumbnail?.href
-        return vals
-      templates:
-        suggestion: (data) ->
-          """
-            <div class='esis-suggestion' \
-                 style='background-image: url(#{data.thumbnail})'>
-            </div>
-            #{data.value}
-          """
-      selected: @onSelect
-    _.defer -> $el.focus()
-
-  onSelect: (e, selected) ->
-    new Artwork(id: selected.id).fetch
-      success: (artwork) =>
-        newImages = @state.images.concat [artwork.denormalized()]
-        @setState images: newImages
-        @props.section.set images: newImages
-        $(@refs.autocomplete).val('').focus()
-        @toggleFillwidth() if @state.images.length > 1
+      @onChange()
+      @$list.animate({opacity: 1}, 250) if @props.section.get('layout') != 'column_width'
 
-  upload: (e) ->
-    gemup e.target.files[0],
-      app: sd.GEMINI_APP
-      key: sd.GEMINI_KEY
-      progress: (percent) =>
-        @setState progress: percent
-      add: (src) =>
-        @$list.animate({opacity: 0}, 500)
-        @setState progress: 0.1
-      done: (src) =>
-        image = new Image()
-        image.src = src
-        image.onload = =>
-          newImages = @state.images.concat [{
-            url: src
-            type: 'image'
-            width: image.width
-            height: image.height
-          }]
-          @setState images: newImages
-          @props.section.set images: newImages
-          imagesLoaded @$list, =>
-            if @state.images.length > 1
-              @toggleFillwidth()
-            else
-              @setState progress: null
-              @$list.animate({opacity: 1}, 500)
-
-  changeLayout: (layout) -> =>
-    @setState layout: layout
-    @props.section.set layout: layout
-    if layout is 'overflow_fillwidth' and @state.images.length > 1
-      @toggleFillwidth()
+  onChange: ->
+    @forceUpdate() if @$list.children().length isnt @props.section.get('images').length
+    if @props.section.get('images').length > 1 and @props.section.get('layout') != 'column_width'
+      @fillwidth()
```

Comment:
Currently, it's [inside Reaction Force](https://github.com/artsy/reaction-force/blob/master/src/components/artwork/fillwidth.tsx), so not for this PR but down the line it might be worth pairing with him or asking in [#collector-gmv-dev](https://artsy.slack.com/messages/C0GNAB0US) if someone would pair on it.